pop11_grammar_ebnf.txt

```diff
@@ -83,25 +87,25 @@ Jump ::=
 	'goto' Word
 
 Iterative ::=
-	'for' Expression 'step' Expression 'till' Expression 'do' Sequence 'endfor'
-	| 'while' Expression 'do' Sequence 'endwhile'
-	| 'until' Expression 'do' Sequence 'enduntil'
-	| 'for' Word (('in' | 'on') | ('from' Expression ('by' Expression)?)? 'to') Expression 'do' Sequence 'endfor'
-	| 'for' Word 'from_repeater' Expression 'do' Sequence 'endfor'
-	| 'foreach' Expression ('in' Expression)? 'do' Sequence 'endforeach'
-	| 'forevery' Expression ('in' Expression)? 'do' Sequence 'endforevery'
-	| 'repeat' (Expression 'times' | 'forever')? Sequence 'endrepeat'
+	'for' NonEmptyExpression 'step' NonEmptyExpression 'till' NonEmptyExpression 'do' Sequence 'endfor'
+	| 'while' NonEmptyExpression 'do' Sequence 'endwhile'
+	| 'until' NonEmptyExpression 'do' Sequence 'enduntil'
+	| 'for' Word (('in' | 'on') | ('from' NonEmptyExpression ('by' NonEmptyExpression)?)? 'to') NonEmptyExpression 'do' Sequence 'endfor'

```

Comment:
Should be a `Variable`, not a `Word`. In addition two of the three expressions can be empty.  The `by` expression must be non-empty (!!).
```
 Setpop
 : [a b c];
 : lvars i; for i in do i => endfor;
 ** a
 ** b
 ** c
 : 
```
Also:
```
 Setpop
 : 4, 1; lvars i; for i from by 2 to do i => endfor;
 ** 1
 ** 3
```